src/python/pants/backend/python/providers/pyenv/rules.py

```diff
@@ -0,0 +1,182 @@
+# Copyright 2023 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from textwrap import dedent
+
+from pants.backend.python.subsystems.setup import PythonSetup
+from pants.backend.python.util_rules.pex import PythonProvider
+from pants.backend.python.util_rules.pex import rules as pex_rules
+from pants.backend.python.util_rules.pex_environment import PythonExecutable
+from pants.core.util_rules.external_tool import (
+    DownloadedExternalTool,
+    ExternalToolRequest,
+    TemplatedExternalTool,
+)
+from pants.core.util_rules.external_tool import rules as external_tools_rules
+from pants.engine.env_vars import EnvironmentVars, EnvironmentVarsRequest
+from pants.engine.fs import CreateDigest, FileContent
+from pants.engine.internals.native_engine import Digest, MergeDigests
+from pants.engine.internals.selectors import Get, MultiGet
+from pants.engine.platform import Platform
+from pants.engine.process import Process, ProcessCacheScope, ProcessResult
+from pants.engine.rules import collect_rules, rule
+from pants.engine.unions import UnionRule
+from pants.option.option_types import StrListOption
+from pants.util.frozendict import FrozenDict
+from pants.util.meta import classproperty
+
+
+class PyenvSubsystem(TemplatedExternalTool):
+    options_scope = "pyenv"
+    name = "pyenv"
+    help = "pyenv (https://github.com/pyenv/pyenv)."
+
+    default_version = "2.3.13"
+    default_url_template = "https://github.com/pyenv/pyenv/archive/refs/tags/v{version}.tar.gz"
+
+    python_configure_opts = StrListOption(
+        help="Flags to use when configuring CPython.",
+        advanced=True,
+    )
+
+    @classproperty
+    def default_known_versions(cls):
+        return [
+            "|".join(
+                (
+                    cls.default_version,
+                    plat,
+                    "9105de5e5cf8dc0eca2a520ed04493d183128d46a2cfb402d4cc271af1bf144b",
+                    "749323",
+                )
+            )
+            for plat in ["macos_arm64", "macos_x86_64", "linux_x86_64", "linux_arm64"]
+        ]
+
+    def generate_exe(self, plat: Platform) -> str:
+        """Returns the path to the tool executable.
+
+        If the downloaded artifact is the executable itself, you can leave this unimplemented.
+
+        If the downloaded artifact is an archive, this should be overridden to provide a
+        relative path in the downloaded archive, e.g. `./bin/protoc`.
+        """
+        return f"./pyenv-{self.version}/bin/pyenv"
+
+
+class PythonToolchainRequest(PythonProvider):
+    pass
+
+
+PYENV_APPEND_ONLY_CACHES = FrozenDict({"pyenv": ".pyenv"})
```

Comment:
would like to see the use of `.pyenv` codified into a CONST or similar to a) make the dependency clear where this is used and b) to avoid inadvertent changes to a subset of them.